src/token/erc20.sol

```diff
@@ -66,78 +57,186 @@ contract ERC20 {
         );
     }
 
-    // --- ERC20 ---
-    function transfer(address dst, uint256 wad) external returns (bool) {
-        return transferFrom(msg.sender, dst, wad);
+    function DOMAIN_SEPARATOR() external view returns (bytes32) {
+        return block.chainid == deploymentChainId ? _DOMAIN_SEPARATOR : _calculateDomainSeparator(block.chainid);
+    }
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "not-authorized");
+        _;
     }
 
-    function transferFrom(address src, address dst, uint256 wad) public virtual returns (bool) {
-        require(balanceOf[src] >= wad, "cent/insufficient-balance");
-        if (src != msg.sender && allowance[src][msg.sender] != type(uint256).max) {
-            require(allowance[src][msg.sender] >= wad, "cent/insufficient-allowance");
-            allowance[src][msg.sender] = safeSub_(allowance[src][msg.sender], wad);
+    // --- Administration ---
+    function rely(address user) external auth {
+        wards[user] = 1;
+        emit Rely(user);
+    }
+
+    function deny(address user) external auth {
+        wards[user] = 0;
+        emit Deny(user);
+    }
+
+    function file(bytes32 what, string memory data) external auth {
+        if (what == "name") name = name;
+        else if (what == "symbol") symbol = symbol;
+        else revert("file-unrecognized-param");
```

Comment:
Do we need to prefix the revert message?

```suggestion
        else revert("Erc20/file-unrecognized-param");
```

Reply:
Yeah makes sense, will change